### PR TITLE
chore(quality): fix code coverage reports by adding svelte files

### LINKED ISSUE #1513: code coverage is not reported on files without tests
### Is your enhancement related to a problem? Please describe

we have code coverage from the current files being tested but untested files are not reported in the report
it's leading to invalid results

### Describe the solution you'd like

Full code coverage report should report coverage for all the files, even those that have no associated tests

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by odockal**
- closes #1513
- change reporter to lcov (preparation for sonar analysis)
- include svelte files into analysis

### What does this PR do?
Improves and fixes issues in code coverage reporting

### What issues does this PR fix or reference?
#1513 

### How to test this PR?
`yarn test`

`yarn test` should create reports in project root in `test-resources/coverage` folder. They are divided by folders the code resides in the project (main, renderer, extensions, etc.). You should find index.html report that is quite detailed for each folder.

### CODE REVIEW

File: extensions/vitest.config.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+/**********************************************************************
+ * Copyright (C) 2022 Red Hat, Inc.
```
Comment (benoitf) [inline extensions/vitest.config.js]:
```suggestion
 * Copyright (C) 2023 Red Hat, Inc.
```

File: extensions/vitest.config.js [hunk 2]
```diff
@@ -0,0 +1,82 @@
+/**********************************************************************
```
Comment (benoitf) [inline extensions/vitest.config.js]: note sure to get why we need a vitest config file in this directory.
this directory contains nothing, only subdirectories

Comment (odockal) [reply extensions/vitest.config.js]: Coverage setup/config from vitest.config.ts file that is in root folder is used when `test:extensions` is being run. It seems a bit confusing to me then, we have config in the root that is applied for extensions (for tests and coverage) and then we have individual vitest.config for packages/subfolders. I could either use the root vitest config and specify coverage config to be used for extensions subfolder, or put individual vitest.config into extensions folder. If I keep it like it is, it produces empty coverage for packages folders as part of extensions coverage report (individual coverage for package/main,etc is ok in its reports).

Comment (benoitf) [reply extensions/vitest.config.js]: I think the idea is that extensions are kind of self-contained

I usually go inside an extension folder like `extensions/compose` and call a command to get code coverage for the extension using the vitest `watch` mode. So I can develop and write the tests and see live the coverage directly inside the IDE.

So I would expect to be able to do that easily. I'm not against adding files in extensions/folder but maybe we need an include from each extension to a common shared config file as well.

Comment (odockal) [reply extensions/vitest.config.js]: Sounds great, will do.

File: extensions/crc/vitest.config.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import path from 'node:path';
+import { coverageConfig } from '../kind/vitest.config';
```
Comment (benoitf) [inline extensions/crc/vitest.config.js]: it seems that there is something wrong as it includes stuff from the kind extension.
extension are not related so they can't import stuff between them
shared file should be in a shared folder but not in an extension

Comment (odockal) [reply extensions/crc/vitest.config.js]: True!

**Comment by odockal** [comment]
@benoitf 
There is more, unfortunately, c8 or vitest-c8 integration works with coverage in specific module/folder in a way, that if there are not test files, no coverage is reported (not even zero coverage for all ie. *.ts files). It is kinda bug/feature I would say, but let's see the reaction from vitest team, I have opened an [issue] there. 
There are two option: first one is to move vitest.config to the parent folder (so that if there is one test file, coverage will be reported for all extensions), which is previous solution. Or to include at least one basic/dummy unit test into each extension. Or third one could be to just stick with actual reports missing.

[issue]: https://github.com/vitest-dev/vitest/issues/3041

**Comment by benoitf** [comment]
@odockal then could the shared file could be in the root folder like `vitest-extension.config.js` ?
We should not add any sibling folder in the `extensions/` folder that is not an extension

we can have :
`extensions/vitest-shared-extensions.config.js`
`vitest-shared-extensions.config.js`
`.config/vitest-shared-extensions.config.js`

and then
`extensions/kind/vitest.config.js` including a shared parent config file from any parent location.


but not a new sibling in `extensions/` like `extensions/config/vitest.config.js`

**Comment by odockal** [comment]
@benoitf review feedback included. This is part of output of `yarn test:extensions`:
```
> podman-desktop@0.13.0-next test:extensions:podman
> vitest run -r extensions/podman --passWithNoTests --coverage


 RUN  v0.29.3 /home/odockal/git/podman-desktop/extensions/podman
      Coverage enabled with c8

stderr | src/compatibility-mode.spec.ts > darwin: DarwinSocketCompatibility class, test runSudoMacHelperCommand ran within runCommand
Error running podman-mac-helper: TypeError: __vite_ssr_import_0__.window.showInformationMessage is not a function

stderr | builtin/podman.cdix/src/compatibility-mode.spec.ts > darwin: DarwinSocketCompatibility class, test runSudoMacHelperCommand ran within runCommand
Error running podman-mac-helper: TypeError: __vite_ssr_import_0__.window.showInformationMessage is not a function

 ✓ src/compatibility-mode.spec.ts (4)
 ✓ builtin/podman.cdix/src/compatibility-mode.spec.ts (4)
 ✓ src/macos-check.spec.ts (3)
 ✓ builtin/podman.cdix/src/macos-check.spec.ts (3)

 Test Files  4 passed (4)
      Tests  14 passed (14)
   Start at  11:23:42
   Duration  679ms (transform 280ms, setup 0ms, collect 461ms, tests 34ms)

 % Coverage report from c8
--------------------------|---------|----------|---------|---------|--------------------------------------------
File                      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s                          
--------------------------|---------|----------|---------|---------|--------------------------------------------
All files                 |   10.29 |    56.25 |    22.8 |   10.29 |                                            
 scripts                  |       0 |        0 |       0 |       0 |                                            
  download.ts             |       0 |        0 |       0 |       0 | 1-312                                      
 src                      |   11.67 |    58.06 |   23.21 |   11.67 |                                            
  base-check.ts           |   94.28 |       60 |     100 |   94.28 | 27-28                                      
  compatibility-mode.ts   |   75.19 |    83.33 |   63.63 |   75.19 | 36-38,52,54,62-65,69-89,117-118            
  detection-checks.ts     |       0 |        0 |       0 |       0 | 1-45                                       
  extension.ts            |       0 |        0 |       0 |       0 | 1-758                                      
  macos-checks.ts         |   54.94 |     62.5 |      25 |   54.94 | 27,29-36,40,43-50,69-91                    
  podman-cli.ts           |   31.85 |      100 |       0 |   31.85 | 26-38,41-51,56-57,65-119,125-135           
  podman-configuration.ts |       0 |        0 |       0 |       0 | 1-235                                      
  podman-install.ts       |       0 |        0 |       0 |       0 | 1-501                                      
  registry-setup.ts       |       0 |        0 |       0 |       0 | 1-199                                      
  util.ts                 |   45.45 |      100 |       0 |   45.45 | 26-27,30-31,34-35,38-39,45-48,61-68,71-110 
  warnings.ts             |       0 |        0 |       0 |       0 | 1-101                                      
--------------------------|---------|----------|---------|---------|--------------------------------------------

> podman-desktop@0.13.0-next test:extensions:registries
> vitest run -r extensions/registries --passWithNoTests --coverage


 RUN  v0.29.3 /home/odockal/git/podman-desktop/extensions/registries
      Coverage enabled with c8

include: **/*.{test,spec}.{js,mjs,cjs,ts,mts,cts,jsx,tsx}
exclude:  **/node_modules/**, **/dist/**, **/cypress/**, **/.{idea,git,cache,output,temp}/**, **/{karma,rollup,webpack,vite,vitest,jest,ava,babel,nyc,cypress,tsup,build}.config.*
watch exclude:  **/node_modules/**, **/dist/**
No test files found, exiting with code 0

Done in 87.55s.
```

second part if that there is not test file and so report shows also no coverage data.

**Review by benoitf — COMMENTED** [review]
are we still able to run all the tests of all the extensions doing
`vitest run -r extensions` ?

or it requires to run in each extension separately ?

**Comment by odockal** [comment]
> 

@benoitf it is possible to run them all in `test:extensions` target. It contains all separate extensions test npm [commands]

[commands]: https://github.com/containers/podman-desktop/pull/1696/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R46

**Comment by benoitf** [comment]
@odockal yes I saw that there are additional tasks, but what I was asking is if we can still run vitest on the parent directory of the extensions instead of having 5 sequentials run (if we have 5 extensions)


I go to root directory of podman-desktop and execute for example `vitest run -r extensions`
will it run all tests from all the extensions ?

**Comment by odockal** [comment]
> I go to root directory of podman-desktop and execute for example vitest run -r extensions
will it run all tests from all the extensions ?

Let me check.

**Comment by odockal** [comment]
@benoitf running this `vitest run -r extensions` via yarn results in this:
```
[odockal@fedora podman-desktop]$ yarn test:extensions:tests
yarn run v1.22.19
$ vitest run -r extensions

 RUN  v0.29.3 /home/odockal/git/podman-desktop/extensions

stdout | compose/src/compose-extension.spec.ts > deactivate
stopping compose extension

stderr | podman/src/compatibility-mode.spec.ts > darwin: DarwinSocketCompatibility class, test runSudoMacHelperCommand ran within runCommand
Error running podman-mac-helper: TypeError: __vite_ssr_import_0__.window.showInformationMessage is not a function

 ✓ compose/src/os.spec.ts (3)
 ✓ compose/src/detect.spec.ts (10) 522ms
 ✓ compose/src/compose-extension.spec.ts (16)
 ✓ compose/src/compose-github-releases.spec.ts (10)
 ✓ podman/src/compatibility-mode.spec.ts (4)
 ✓ kind/src/kind-installer.spec.ts (1)
 ✓ compose/src/compose-wrapper-generator.spec.ts (4)
 ✓ podman/src/macos-check.spec.ts (3)

 Test Files  8 passed (8)
      Tests  51 passed (51)
   Start at  12:28:59
   Duration  1.69s (transform 1.13s, setup 2ms, collect 1.83s, tests 675ms)

Done in 2.10s.
```

So the answer seems to be "yes".

**Review by benoitf — APPROVED** [review]

**Comment by odockal** [comment]
Follow up issue #1759.

**Comment by benoitf** [comment]
it looks like the yarn.lock file is corrupted (should not be part of your PR anyway)

**Comment by odockal** [comment]
@benoitf Should be fixed now.

**Review by benoitf — APPROVED** [review]
I checked that I have .svelte code coverage
image


as a follow-up we could also ensure that there is a `test` task inside each `package.json` of an extension

If I go to an extension folder and launch `yarn test` it should work

https://github.com/containers/podman-desktop/blob/main/extensions/podman/package.json#L91

(but here there is no code coverage enabled by default) and some other extensions are missing the `test` task

**Comment by odockal** [comment]
@benoitf here we go: https://github.com/containers/podman-desktop/issues/1766

**Pull request merged**